feat/#issue18, implemented functionality to list all saved commits in...

Marcus9512 opened the pull request:
ascending order (with the lowest timestamp first). The commit also
contains an implementation of a method that search for a commit and
returns the file if it exists, otherwise null.

All implmented methods have a test in HistoryTest class the checks the
functionality.

Marcus9512 committed:
feat/#issue18, implemented functionality to list all saved commits in
ascending order (with the lowest timestamp first). The commit also
contains an implementation of a method that search for a commit and
returns the file if it exists, otherwise null.

All implmented methods have a test in HistoryTest class the checks the
functionality.

georgewbar commented on src/main/java/History.java:
```diff
@@ -1,49 +1,167 @@
+     * Initialize history with a path to test folder. This mode is used if the class should be tested as a JUnit test.
+     *
+     * @param isThisATestCase, if isThisATestCase it true the class will redirect all function to a test folder. This
+     *                         is done to prevent damege on data during tests. If isThisATestCase is false, the class
```
Delete the comma because it does not become formatted correctly in javadoc.

georgewbar commented on src/main/java/History.java:
```diff
@@ -1,49 +1,167 @@
+        if (!file.exists()) {
+            file.mkdir();
+        }
     }
```
This constructor could have called `new History(false);` to prevent code duplication.

Marcus9512 replied:
Sure, but i will fix that in another way, it´s nice to be able to create a class without any parameters (if you only want standard settings)

georgewbar commented on src/main/java/History.java:
```diff
@@ -1,49 +1,167 @@
+        for (File file : files) {
+            String[] info = file.getName().split(" ");
+            if (info.length != 2)
+                return null;
```
Shouldn't this line be `continue;` instead of `return null;` (because the loop should continue searching even if there was an error in one file)?

Marcus9512 replied:
Will fix that, seems better. I added return null to signal that the database is corrupted in some sort of way.

georgewbar commented on src/main/java/History.java:
```diff
@@ -1,49 +1,167 @@
     }
 
     /**
      * The function checks if the file containing the name commiteID exists and return the file, otherwise null
```
The documentation here needs to fixed a little bit.

georgewbar commented on src/test/java/HistoryTest.java:
```diff
@@ -0,0 +1,110 @@
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
If an exception happens, the test case will be successful. Shouldn't this be changed to
```
} catch (IOException e) {
      fail();
}
```

georgewbar commented on src/test/java/HistoryTest.java:
```diff
@@ -0,0 +1,110 @@
+            assertTrue(history.listHistory().compareTo(correct) == 0);
+        } catch (Exception e) {
+            fail();
+            e.printStackTrace();
```
The statement `e.printStackTrace();` is never reached because of `fail()`, isn't it?

georgewbar commented on src/test/java/HistoryTest.java:
```diff
@@ -0,0 +1,110 @@
+        }
+        try {
+            String correct = "test 20T01T20T30\ntest 20T03T19T30\ntest 20T03T20T30\n";
+            assertTrue(history.listHistory().compareTo(correct) == 0);
```
isn't is better to write this line as
`assertEquals(correct, history.listHistory());`
instead, because `correct` is the expected value and `history.listHistory()) == 0` is the actual value?

georgewbar commented on src/test/java/HistoryTest.java:
```diff
@@ -0,0 +1,110 @@
+            writer.close();
+
+        } catch (IOException e) {
+            e.printStackTrace();
```
Shouldn't this line be `fail();` instead?

Marcus9512 replied:
should be both

georgewbar commented on src/test/java/HistoryTest.java:
```diff
@@ -0,0 +1,110 @@
+            assertNull(result);
+        } catch (Exception e) {
+            fail();
+            e.printStackTrace();
```
This line is never reached, isn't it?

georgewbar reviewed (changes requested):
* Delete the commas after paramters `@param ... ,` as in [here](https://github.com/DD2480-Group17/assignment2/blob/b1f77e2ba703db79dc3eb9cc094a3ae29f70fe36/src/main/java/History.java#L28) in the code documentation because it makes javadoc not work properly.
* It is better to have a descending order to see the latest issues and not to have to scroll to the end of the history every time when viewing it from the browser (this is just my opinion and not necessarly true).
* More clarification in the documentation is needed about the format of the timestamp to make it easy to implement the HTML page later.
* Sometime, `compareTo` method is used to see if two strings are equal in both `History.java` class and `HistoryTest.java` class. It is better to use equal when the purpose is to check if two strings are equal. This is especially important in the `HistoryTest.java` class. i.e. it is important to use `assertEquals` when possible to make it easy to know why the test case failed in the test case reports by seeing e.g. `The expected value is <x>, but got <y>`, where `<x>` and `<y>` are different.

Marcus9512 committed:
refactore/#issue18, refactored the code. The listHistory method now
returns the history in decending oreder with the latest commit first.
Additional documentation was added to clarify some methods output.
AssertTrue was changed to assertEquals in some test cases. The position
of fail() was moved behind e.stackTrace.

Marcus9512 committed:
Merge branch 'master' into issue/18

georgewbar commented:
Looks good to me.

georgewbar reviewed (approved):
Looks good to me.
